tests/Yaapii.Atoms.Tests/Number/IsNumberTests.cs

```diff
@@ -0,0 +1,110 @@
+﻿// MIT License
+//
+// Copyright(c) 2019 ICARUS Consulting GmbH
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using Xunit;
+using Yaapii.Atoms.Text;
+
+namespace Yaapii.Atoms.Number.Tests
+{
+    public sealed class IsNumberTests
+    {
+        [Fact]
+        public void ReadsString()
+        {
+            Assert.True(
+                new IsNumber(
+                    "1,234.56"
+                ).Value(),
+                "Can't read number from string"
+            );
+        }
+
+        [Fact]
+        public void ReadsStringWithSpecifiedFormatProvider()
```

Comment:
DetectsCustomCultureNumber

Please consider replacing NumberFormatInfo with IFormatProvider in the class

Reply:
Where exactly? 
The field storing the provider is of type IFormatProvider. 
Consequently, the providers passed to the constructers are also of type IFormatProvider, so that the field can be initialized with them.
The only places where i can see NumberFormatInfo is in secondary constructors, where NumberFormatInfo.InvariantInfo is passed to the primary constructor as an object implementing IFormatProvider.
I couldn't replace NumberFormatInfo.InvariantInfo with IFormatProvider, because i need an object instance there, not just a type.

Reply:
@DFU398 nevermind, I was on the wrong path. Your implementation is okay.